dedupe/variables/datetime_predicates.py

```diff
@@ -18,6 +20,7 @@ def make_predicate(attrs):
     output = tuple(attrs)
     # Check that all resolutions are present in the string
     if all(output):
+        output = tuple(['-'.join(str(dt) for dt in output)])
```

Comment:
Turns out in this case we do need the internal comprehension:

<img width="399" alt="screen shot 2017-06-14 at 9 43 55 am" src="https://user-images.githubusercontent.com/14170650/27138403-0e519ea0-50e6-11e7-9498-86a34eb68205.png">

Which makes sense, because `string.join()` transforms the output of the comprehension into a string.